### PR TITLE
refactor: migrate dialogs to Compose

### PR DESCRIPTION
**Pull request by FunkyMuse**

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/thankyou/helpers/Extensions.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+import android.content.ComponentName
+import android.content.Context
+import android.content.pm.PackageManager
+import androidx.activity.ComponentActivity
+import com.simplemobiletools.commons.extensions.*
+import com.simplemobiletools.commons.models.Release
+import com.simplemobiletools.thankyou.R
+
+fun ComponentActivity.appLaunchedCompose(
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/thankyou/helpers/Extensions.kt:11]: all of this should be in commons as it affects all apps

**Comment by tibbi** [comment]
this app has no Pro version, everything related to that should be in commons. Also Rate Stars dialog in About still has wrong bg.

**Comment by tibbi** [comment]
RateStarsDialog on main screen triggered by simple `RateStarsDialog(this)` still has wrong bg

**Comment by FunkyMuse** [comment]
> RateStarsDialog on main screen triggered by simple `RateStarsDialog(this)` still has wrong bg

That is not a Compose dialog and isn't used here anyway?

**Comment by tibbi** [comment]
it is a compose dialog, it shows on the main screen after 40 app launches

**Comment by tibbi** [comment]
you can reproduce it by adding `baseConfig.appRunCount = 39` just before `appLaunched(...)` in mainactivity

**Comment by FunkyMuse** [comment]
> baseConfig.appRunCount = 39

Which theme?

**Comment by tibbi** [comment]
alright thanks, looks good now

**Comment by tibbi** [comment]
actually, lets properly update the link color of the main screens page in onResume. If you change the primary color from orange the blue and go back, the link is still orange and it becomes blue only after the app restart

**Comment by FunkyMuse** [comment]
> > baseConfig.appRunCount = 39
> 
> Which theme?
> 
> Edit: found it, working on fix

https://github.com/SimpleMobileTools/Simple-Commons/pull/1867

**Comment by FunkyMuse** [comment]
> actually, lets properly update the link color of the main screens page in onResume. If you change the primary color from orange to blue and go back to the main screen, the link is still orange and it becomes blue only after the app restart

https://github.com/SimpleMobileTools/Simple-Commons/pull/1867

**Comment by tibbi** [comment]
one more thing, the app settings language value isnt readable in casual light theme, as it is light grey text on white bg

**Comment by FunkyMuse** [comment]
> one more thing, the app settings language value isnt readable in casual light theme, as it is light grey text on white bg

should be black right?

**Comment by tibbi** [comment]
no, it should have a bit weaker color, it already works well in other themes, check Dark how is it done

**Pull request merged**